numpy/_core/src/umath/loops_hyperbolic.dispatch.cpp.src

```diff
@@ -492,35 +629,38 @@ simd_tanh_f32(const float *src, npy_intp ssrc, float *dst, npy_intp sdst, npy_in
     };
 #endif // defined(TANHF_TRANSPOSED_LUT)
 
-    const int nlanes = npyv_nlanes_f32;
-    const npyv_f32 qnan = npyv_setall_f32(NPY_NANF);
+    const int nlanes = hn::Lanes(f32);//npyv_nlanes_f32;
+    const vec_f32 qnan = hn::Set(f32, NPY_NAN);
     for (; len > 0; len -= nlanes, src += ssrc*nlanes, dst += sdst*nlanes) {
-        npyv_f32 x;
-        if (ssrc == 1) {
-            x = npyv_load_tillz_f32(src, len);
+        vec_f32 x;
+        
+        x = load_vector<vec_f32>(src, ssrc, len);
+        
+        /*if (ssrc == 1) {
+            x = hn::LoadN(f32, src, len);
         } else {
-            x = npyv_loadn_tillz_f32(src, ssrc, len);
-        }
-        npyv_s32 ndnan = npyv_and_s32(npyv_reinterpret_s32_f32(x), npyv_setall_s32(0x7fe00000));
+            x = GatherIndexN32(src, ssrc, len);
+        }*/
+        vec_s32 ndnan = hn::And(hn::BitCast(s32, x), hn::Set(s32, 0x7fe00000));
         // check |x| > HUGE_THRESHOLD, INF and NaNs.
-        npyv_b32 special_m = npyv_cmple_s32(ndnan, npyv_setall_s32(0x7f000000));
-        npyv_b32 nnan_m = npyv_notnan_f32(x);
-        npyv_s32 idxs = npyv_sub_s32(ndnan, npyv_setall_s32(0x3d400000));
-                 idxs = npyv_max_s32(idxs, npyv_zero_s32());
-                 idxs = npyv_min_s32(idxs, npyv_setall_s32(0x3e00000));
-        npyv_u32 idx  = npyv_shri_u32(npyv_reinterpret_u32_s32(idxs), 21);
+        auto special_m = hn::Le(ndnan, hn::Set(s32, 0x7f000000));
+        auto nnan_m = hn::Not(hn::IsNaN(x));
+        vec_s32 idxs = hn::Sub(ndnan, hn::Set(s32, 0x3d400000));
+                 idxs = hn::Max(idxs, hn::Set(s32, 0)); // TODO probably faster zero method
```

Comment:
Yes, you can either use `hn::Max(idxs, hn::Zero(s32))` or even `hn::ZeroIfNegative(idxs)`.